### PR TITLE
Fixing TypeError when external docs exist without servers

### PR DESCRIPTION
**Pull request by philsturgeon**
Ported https://github.com/wework/speccy/pull/52 over.

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Hi @philsturgeon please could you include a failing test-case which you believe would be fixed by this change, so I can see the original issue?

**Comment by philsturgeon** [comment]
Pushed 44a8028 which fails with this:

screen shot 2018-05-01 at 12 25 56 pm

I've then pushed a second commit (29a9f55) with the fix (corrected from your feedback on Speccy). I wish I had more accurate tests, but the validator is not really unit tested.

**Comment by MikeRalphson** [comment]
Thanks @philsturgeon - I think the actual fix is to backport this change from `openapi-kit`:

https://github.com/philsturgeon/swagger2openapi/blob/29a9f5547bcbe26861ab6b599ca8f219e6efdf23/validate.js#L56

Changing

`if (servers.length) {`

to

`if (servers && servers.length) {`

Rather than adding the extra test only at the `externalDocs` site.

File: validate.js [hunk 1]
```diff
@@ -291,7 +291,10 @@ function checkSubSchema(schema, parent, state) {
 
     if (schema.externalDocs) {
         schema.externalDocs.should.have.key('url');
-        should.doesNotThrow(function() { validateUrl(schema.externalDocs.url, [state.openapi.servers], 'externalDocs', state.options)}, 'Invalid externalDocs.url');
+        should.doesNotThrow(function() {
```
Comment (MikeRalphson) [inline validate.js]: This change should no longer be necessary. Sorry.

**Pull request merged**